frontend/src/contexts/AuthContext.tsx

```diff
@@ -36,32 +39,40 @@ const AuthContextProvider: FC<PropsWithChildren<{}>> = props => {
   const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
   const [userId, setUserId] = useState<undefined | string>(undefined);
   const [token, setToken] = useState<undefined | string>(undefined);
+  const { makeErrorCatcher, setExecLogin, execLogin } =
+    useContext(ErrorContext);
+
   useEffect(() => {
-    kc.init({ onLoad: 'login-required' })
-      .then((authenticated: boolean) => {
-        if (authenticated) {
-          setIsLoggedIn(true);
-          setToken(kc.idToken);
-        } else {
-          setIsLoggedIn(false);
-          setToken(undefined);
-          setUserId(undefined);
-        }
-        kc.loadUserInfo()
-          .then((res: any) => {
-            setUserId(res.preferred_username);
-          })
-          .catch(err => {
-            console.error('Error when getting user info', err);
-          });
-      })
-      .catch(err => {
-        console.error('Failed to initialize keycloak', err);
-      });
-  }, []);
+    if (execLogin) {
+      kc.init({ onLoad: 'login-required' })
+        .then((authenticated: boolean) => {
+          if (authenticated) {
+            setIsLoggedIn(true);
+            setToken(kc.idToken);
+          } else {
+            setIsLoggedIn(false);
+            setToken(undefined);
+            setUserId(undefined);
+          }
+          kc.loadUserInfo()
+            .then((res: any) => setUserId(res.preferred_username))
+            .catch(makeErrorCatcher(ErrorTypes.KeycloakError));
+        })
+        .catch(makeErrorCatcher(ErrorTypes.KeycloakError))
+        .finally(() => setExecLogin(false));
+    }
+    //kc.updateToken(30);
```

Comment:
Can this be removed?

Reply:
If you mean the commented line, yes I forgot to delete it.